server/internal/core/application/utils.go

```diff
@@ -270,6 +270,21 @@ func (m *txRequestsQueue) view(id string) (*domain.TxRequest, bool) {
 	}, true
 }
 
+func (m *txRequestsQueue) isVtxoRegisteredForNextRound(vtxoKey domain.VtxoKey) bool {
```

Comment:
this can be expensive if there are a lot of tx requests and end up slowing down the SubmitRedeemTx execution. I think it's better to use another map like `redeemTxRequests` to make this check O(1) instead of O(n)

Reply:
Done